Source/Urho3D/Input/DirectionalPadAdapter.cpp

```diff
@@ -0,0 +1,405 @@
+//
+// Copyright (c) 2022-2022 the rbfx project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include "../Input/DirectionalPadAdapter.h"
+#include "../Core/Context.h"
+#include "../Input/Input.h"
+#include "../Input/InputEvents.h"
+
+namespace Urho3D
+{
+using namespace DirectionalPadAdapterDetail;
+
+/// Construct.
+DirectionalPadAdapter::DirectionalPadAdapter(Context* context)
+    : Object(context)
+    , input_(context->GetSubsystem<Input>())
+{
+    unsigned numJoysticks = input_->GetNumJoysticks();
+    for (unsigned i = 0; i < numJoysticks; ++i)
+    {
+        auto* joystick = input_->GetJoystickByIndex(i);
+        if (joystick->GetNumAxes() == 3 && joystick->GetNumButtons() == 0 && joystick->GetNumHats() == 0)
```

Comment:
This is weird condition that at least deserves a comment. Why there can't be two such joysticks?

Reply:
I relay on knowledge about internal SDL2 implementation details. I can also check for name of the device as SDL2 assigns specific names to it.

At the  moment SDL2 only makes one virtual joystick.